src/pytorch_lightning/utilities/data.py

```diff
@@ -127,14 +127,16 @@ def has_len(dataloader: Union[DataLoader, Iterable]) -> bool:
 
 def has_len_all_ranks(
     dataloader: DataLoader,
-    training_type: "pl.Strategy",
+    training_type: "pl.Strategy",  # type: ignore
     model: Union["pl.LightningModule", "pl.LightningDataModule"],
 ) -> bool:
     """Checks if a given Dataloader has ``__len__`` method implemented i.e. if it is a finite dataloader or
     infinite dataloader."""
     try:
         local_length = len(dataloader)
-        total_length = training_type.reduce(torch.tensor(local_length).to(model.device), reduce_op="sum")
+        total_length = training_type.reduce(
+            torch.tensor(local_length).to(model.device), reduce_op="sum"
+        )  # type: ignore
```

Comment:
Why to ignore here?

Reply:
`model: Union["pl.LightningModule", "pl.LightningDataModule"]`,  `pl.LightningDataModule` does not have any device attribute. What else can I try?